src/csv.ts

```diff
@@ -23,34 +25,52 @@ type EventType = `${Event}`;
 
 type ColumnHeader = { name: string; type: DataType };
 
-export class CSV {
+export class CSV<D extends string> {
     protected _options: CsvLoaderOptions;
     protected _loader: Loader;
-    protected _handlers: Map<Event, Set<EventHandler>>;
+    protected _handlers: Map<EventType, Map<D, Set<EventHandler>>>;
+
+    #openedDataSource: D;
 
-    public constructor(options: CsvLoaderOptions) {
-        this._options = Object.assign({}, options);
+    public constructor(options: Partial<CsvLoaderOptions>) {
+        this._options = {
+            dataSources: {},
+            includesHeader: true,
+            typeInferLines: 20,
+            verbose: false,
+            ...options,
+        };
         this._loader = new Loader();
         this._loader.onOpened = this.dispatch.bind(this, Event.Opened);
         this._loader.onColumns = this.dispatch.bind(this, Event.Columns);
         this._loader.onData = this.dispatch.bind(this, Event.Data);
         this._loader.onDone = this.dispatch.bind(this, Event.Done);
         this._loader.onError = this.dispatch.bind(this, Event.Error);
-        this._handlers = new Map<Event, Set<EventHandler>>([
-            [Event.Opened, new Set<EventHandler>()],
-            [Event.Columns, new Set<EventHandler>()],
-            [Event.Data, new Set<EventHandler>()],
-            [Event.Done, new Set<EventHandler>()],
-            [Event.Error, new Set<EventHandler>()],
+        this._handlers = new Map<EventType, Map<D, Set<EventHandler>>>([
```

Comment:
In my opinion, this nested something is some weird stuff.
PR #9 proposes a solution for this.